src/tests/unit/platform/test_users.py

```diff
@@ -0,0 +1,217 @@
+    api = cbcsdk_mock.api
+    user = api.select(User, 4338)
+    assert user.last_name == 'Jackson'
+    assert user.first_name == 'Daniel'
+    assert user.login_name == 'djackson@stargate-command.af.mil'
+    assert user.email == 'djackson@stargate-command.af.mil'
+    assert user.urn == 'psc:user:test:4338'
+    user.reset_google_authenticator_registration()
+
+
+def test_get_nonexistent_user(cbcsdk_mock):
+    """Test retrieving a nonexistent user."""
+    cbcsdk_mock.mock_request('GET', '/appservices/v6/orgs/test/users', GET_USERS_RESP)
+    api = cbcsdk_mock.api
+    with pytest.raises(ObjectNotFoundError):
+        api.select(User, 999)
+
+
+def test_attempt_to_create_naively(cb):
+    """Attempt to create a user by just creating one and saving it, which doesn't work."""
+    user = User(cb, None)
+    user.last_name = 'Rios'
+    user.first_name = 'Cristobal'
+    user.login_name = 'rios@la-sirena.net'
+    user.email = 'rios@la-sirena.net'
+    with pytest.raises(ApiError):
+        user.save()
+
+
+def test_create_user(cbcsdk_mock):
+    """Test creating a user."""
+    post_made = False
+
+    def check_post(uri, body, **kwargs):
+        assert body == EXPECT_USER_ADD
+        nonlocal post_made
+        post_made = True
+        return USER_ADD_SUCCESS_RESP
+
+    def check_get(uri, query_params, default):
+        nonlocal post_made
+        assert post_made
+        return GET_USERS_AFTER_CREATE_RESP
+
+    cbcsdk_mock.mock_request('POST', '/appservices/v6/orgs/test/users', check_post)
+    cbcsdk_mock.mock_request('GET', '/appservices/v6/orgs/test/users', check_get)
+    api = cbcsdk_mock.api
+    builder = User.create(api).set_email('rios@la-sirena.net').set_role('psc:role:test:APP_SERVICE_ROLE')
+    builder.set_first_name('Cristobal').set_last_name('Rios')
+    builder.add_grant_profile(['psc:org:test2'], ['psc:role:test2:DUMMY'])
+    builder.add_grant_profile(['test3'], ['psc:role:test3:DUMMY'])
+    user, password = builder.build()
+    assert post_made
+    assert password == 'abcd_efgh'
+    assert user.last_name == 'Rios'
+    assert user.first_name == 'Cristobal'
+    assert user.login_name == 'rios@la-sirena.net'
+    assert user.email == 'rios@la-sirena.net'
+
+
+def test_create_user_from_template(cbcsdk_mock):
+    """Test creating a user from a template."""
+    post_made = False
+
+    def check_post(uri, body, **kwargs):
+        assert body == EXPECT_USER_ADD_SMALL
+        nonlocal post_made
+        post_made = True
+        return USER_ADD_SUCCESS_RESP
+
+    def check_get(uri, query_params, default):
+        nonlocal post_made
+        assert post_made
+        return GET_USERS_AFTER_CREATE_RESP
+
+    cbcsdk_mock.mock_request('POST', '/appservices/v6/orgs/test/users', check_post)
+    cbcsdk_mock.mock_request('GET', '/appservices/v6/orgs/test/users', check_get)
+    api = cbcsdk_mock.api
+    template = {
+        "email_id": "rios@la-sirena.net",
+        "role_urn": "psc:role:test:APP_SERVICE_ROLE",
```

Comment:
A grant can't have both a role_urn and a profile, it's one or the other

Reply:
This is _test code,_ intended to see that the data gets slotted-in correctly.  Does that matter? If it does, I'll have to write extra tests.

Reply:
Test code must follow the same usage patterns otherwise this may cause confusion in the future or a developer may look at test code and make a wrong assumption

Reply:
In my opinion, this adds _nothing_ to the tests, only increasing complexity for no good reason. It does not increase code coverage of the tests, for instance.  Nevertheless, I will do as you ask.

Reply:
This is not intended to add anything to the tests from a functional stand point instead this maintains the perspective of how to use the SDK correctly. If we take short cuts and perform tests which will through an API error then our tests don't match to the functionality of the developers experience and is therefore an invalid testcase